src/engraving/libmscore/edit.cpp

```diff
@@ -4830,6 +4830,9 @@ static EngravingItem* findLinkedVoiceElement(EngravingItem* e, Staff* nstaff)
     Measure* measure = segment->measure();
     Measure* m       = score->tick2measure(measure->tick());
     Segment* s       = m->findSegment(segment->segmentType(), segment->tick());
+    if (!s) {
```

Comment:
This looks very suspicious. Where do these null pointers come from? It looks like this PR fixes the consequence of the problem, not the cause

Reply:
Well, I don't quite understand the question, obviously this one comes from `findSegment()`.

Reply:
That in turn used `findSegmentR()`
```
//---------------------------------------------------------
//   findSegmentR
//    Search for a segment of type st at measure relative
//    position t.
//---------------------------------------------------------

Segment* Measure::findSegmentR(SegmentType st, const Fraction& t) const
{
    Segment* s;
    if (t > (ticks() * Fraction(1, 2))) {
        // search backwards
        for (s = last(); s && s->rtick() > t; s = s->prev()) {
        }
        while (s && s->prev() && s->prev()->rtick() == t) {
            s = s->prev();
        }
    } else {
        // search forwards
        for (s = first(); s && s->rtick() < t; s = s->next()) {
        }
    }
    for (; s && s->rtick() == t; s = s->next()) {
        if (s->segmentType() & st) {
            return s;
        }
    }
    return 0;
}
```

Reply:
Why does it return null in this case? It usually indicates some other, more complicated problem

Reply:
Dunno. The method does on occasions and on purpose return a `nullptr`. So callers should take care of those and not relying on it to never do that.

Reply:
If it were illegal, an assert would be due in that `findSegmentR()`